internal/chain/chain.go

```diff
@@ -0,0 +1,83 @@
+package chain
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/node/engine/chainservice"
+	NitroAdjudicator "github.com/statechannels/go-nitro/node/engine/chainservice/adjudicator"
+	ConsensusApp "github.com/statechannels/go-nitro/node/engine/chainservice/consensusapp"
+	chainutils "github.com/statechannels/go-nitro/node/engine/chainservice/utils"
+	VirtualPaymentApp "github.com/statechannels/go-nitro/node/engine/chainservice/virtualpaymentapp"
+	"github.com/statechannels/go-nitro/types"
+)
+
+const (
+	FUNDED_TEST_PK  = "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+	ANVIL_CHAIN_URL = "ws://127.0.0.1:8545"
+)
+
+type ChainOpts struct {
+	ChainUrl       string
+	ChainAuthToken string
+	ChainPk        string
+	NaAddress      common.Address
+	VpaAddress     common.Address
+	CaAddress      common.Address
+}
+
+func InitializeEthChainService(chainOpts ChainOpts) (*chainservice.EthChainService, error) {
+	if chainOpts.ChainPk == "" {
+		return nil, fmt.Errorf("chainpk must be set")
+	}
+
+	fmt.Println("Initializing chain service and connecting to " + chainOpts.ChainUrl + "...")
+	return chainservice.NewEthChainService(
+		chainOpts.ChainUrl,
+		chainOpts.ChainAuthToken,
+		chainOpts.ChainPk,
+		chainOpts.NaAddress,
+		chainOpts.CaAddress,
+		chainOpts.VpaAddress,
+		os.Stdout)
+}
+
+func StartAnvil() (*exec.Cmd, error) {
+	chainCmd := exec.Command("anvil", "--chain-id", "1337")
+	chainCmd.Stdout = os.Stdout
+	chainCmd.Stderr = os.Stderr
+	err := chainCmd.Start()
+	if err == nil {
+		// Give the chain a second to start up
+		time.Sleep(1 * time.Second)
+	}
+	return chainCmd, err
```

Comment:
I'm a bit confused by this error handling. If there is no error, we wait for 1s. If there is an error, it will be returned, but with a non-nil `chainCmd.

Reply:
I see. I think it is valid to return a non-nil `chainCmd`, but I am not sure wha the convention is. I think almost everywhere else we return the default value alongside non-nil errors? I don't feel super strongly about it. 

More nitpicking -- it is also a bit unusual for us to do a check for nil rather than not nil.

```go
	if err != nil {
                 return nil, err
        }
        time.Sleep(1 * time.Second) // give time for chain to start up
	return chainCmd, nil
```

Reply:
These are good points. Refactored in 499a4cec.